js/test/test.js

```diff
@@ -342,39 +382,64 @@ async function testExchange (exchange) {
         await test ('InvalidOrder', exchange, symbol);
         await test ('InsufficientFunds', exchange, symbol, balance); // danger zone - won't execute with non-empty balance
     }
+
+    await test ('addMargin', exchange, symbol);
+    await test ('reduceMargin', exchange, symbol);
+    await test ('setMargin', exchange, symbol);
+    await test ('setMarginMode', exchange, symbol);
+    await test ('setPositionMode', exchange, symbol);
+    await test ('setLeverage', exchange, symbol);
+    await test ('cancelAllOrders', exchange, symbol);
+    await test ('cancelOrder', exchange, symbol);
+    await test ('cancelOrders', exchange, symbol);
+    await test ('fetchCanceledOrders', exchange, symbol);
+    await test ('fetchClosedOrder', exchange, symbol);
+    await test ('fetchOpenOrder', exchange, symbol);
+    await test ('fetchOrder', exchange, symbol);
+    await test ('fetchOrderTrades', exchange, symbol);
+    await test ('fetchPosition', exchange, symbol);
+    await test ('fetchFundingHistory', exchange, symbol);
+    await test ('fetchDeposit', exchange, code);
+    await test ('createDepositAddress', exchange, code);
+    await test ('fetchDepositAddress', exchange, code);
+    await test ('fetchDepositAddresses', exchange, code);
+    await test ('fetchDepositAddressesByNetwork', exchange, code);
+    await test ('editOrder', exchange, symbol);
+    await test ('fetchBorrowRateHistory', exchange, symbol);
+    await test ('fetchBorrowRatesPerSymbol', exchange, symbol);
+    await test ('fetchLedgerEntry', exchange, code);
+    await test ('fetchPositionsRisk', exchange, symbol);
+    await test ('fetchWithdrawal', exchange, code);
+    await test ('transfer', exchange, code);
+    await test ('withdraw', exchange, code);
 }
 
 //-----------------------------------------------------------------------------
 
 async function tryAllProxies (exchange, proxies) {
 
-    const index = proxies.indexOf (exchange.proxy);
+    const index = findValueIndexInArray (proxies, exchange.proxy);
```

Comment:
maybe we can remove proxies at all from here? if our tests run normally, then "proxies testing" might not be our duty and proxies-compatibility should be destined for userlands.

also, if proxies is still actual, then we we can just pass 'proxy' argument somehow to tests, so the tests will run on it's own, the only diff will be that it will use the provided 'proxy' internally, and this hardcoded method will not be needed.

Reply:
@ttodua I don't fully understand this either, we should only use a cors proxy when running stuff on the browser I think which is not the case here